tests/components/homematicip_cloud/test_init.py

```diff
@@ -1,154 +1,115 @@
+
+    # one config_entry exists
+    config_entries = hass.config_entries.async_entries(hmipc.DOMAIN)
+    assert len(config_entries) == 1
+    assert config_entries[0].data == {
+        "authtoken": "123",
+        "hapid": "ABC123",
+        "name": "name",
+    }
+    # config_enty has no unique_id
+    assert not config_entries[0].unique_id
 
-async def test_setup_entry_successful(hass):
-    """Test setup entry is successful."""
-    entry = MockConfigEntry(
-        domain=hmipc.DOMAIN,
-        data={
-            hmipc.HMIPC_HAPID: "ABC123",
-            hmipc.HMIPC_AUTHTOKEN: "123",
-            hmipc.HMIPC_NAME: "hmip",
-        },
+    entry_config = {
+        hmipc.CONF_ACCESSPOINT: "ABC123",
+        hmipc.CONF_AUTHTOKEN: "123",
+        hmipc.CONF_NAME: "name",
+    }
+    assert (
+        await async_setup_component(hass, hmipc.DOMAIN, {hmipc.DOMAIN: entry_config})
+        is True
     )
-    entry.add_to_hass(hass)
-    with patch.object(hmipc, "HomematicipHAP") as mock_hap:
-        instance = mock_hap.return_value
-        instance.async_setup.return_value = mock_coro(True)
-        instance.home.id = "1"
-        instance.home.modelType = "mock-type"
-        instance.home.name = "mock-name"
-        instance.home.currentAPVersion = "mock-ap-version"
-
-        assert (
-            await async_setup_component(
-                hass,
-                hmipc.DOMAIN,
-                {
-                    hmipc.DOMAIN: {
-                        hmipc.CONF_ACCESSPOINT: "ABC123",
-                        hmipc.CONF_AUTHTOKEN: "123",
-                        hmipc.CONF_NAME: "hmip",
-                    }
-                },
-            )
-            is True
-        )
-
-    assert len(mock_hap.mock_calls) >= 2
-
-
-async def test_setup_defined_accesspoint(hass):
-    """Test we initiate config entry for the accesspoint."""
-    with patch.object(hass, "config_entries") as mock_config_entries, patch.object(
-        hmipc, "configured_haps", return_value=[]
-    ):
-        mock_config_entries.flow.async_init.return_value = mock_coro()
-        assert (
-            await async_setup_component(
-                hass,
-                hmipc.DOMAIN,
-                {
-                    hmipc.DOMAIN: {
-                        hmipc.CONF_ACCESSPOINT: "ABC123",
-                        hmipc.CONF_AUTHTOKEN: "123",
-                        hmipc.CONF_NAME: "hmip",
-                    }
-                },
-            )
-            is True
-        )
 
-    assert len(mock_config_entries.flow.mock_calls) == 1
-    assert mock_config_entries.flow.mock_calls[0][2]["data"] == {
-        hmipc.HMIPC_HAPID: "ABC123",
-        hmipc.HMIPC_AUTHTOKEN: "123",
-        hmipc.HMIPC_NAME: "hmip",
+    # no new config_entry created / still one config_entry
+    config_entries = hass.config_entries.async_entries(hmipc.DOMAIN)
+    assert len(config_entries) == 1
+    assert config_entries[0].data == {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4887152)